src/ngraph_builder.cc

```diff
@@ -3264,35 +3264,54 @@ static Status TranslateSplitVOp(
     const Node* op, const std::vector<const Tensor*>& static_input_map,
     Builder::OpMap& ng_op_map) {
   shared_ptr<ng::Node> ng_input, ng_length, ng_split_dim;
-  TF_RETURN_IF_ERROR(
-      GetInputNodes(ng_op_map, op, &ng_input, &ng_length, &ng_split_dim));
+  TF_RETURN_IF_ERROR(GetInputNode(ng_op_map, op, 0, &ng_input));
 
   std::vector<int> lengths;
   TF_RETURN_IF_ERROR(GetStaticInputVector(op, 1, static_input_map, &lengths));
 
+  int length = 0;
+  int shape_length = 0;
+  int idx;
+  for (int i = 0; i < lengths.size(); ++i) {
+    if (lengths[i] != -1) {
+      length += lengths[i];
+    } else {
+      idx = i;
+    }
+  }
+
   ng::Shape shape = ng_input->get_shape();
   int rank = shape.size();
-  std::vector<size_t> lower;
-  std::vector<size_t> upper;
-
-  for (int i = 0; i < rank; ++i) {
-    lower.push_back(0);
-    upper.push_back(shape[i]);
+  for (int i = 0; i < shape.size(); ++i) {
+    shape_length += shape[i];
+  }
+  if (shape_length > length) {
+    lengths[idx] = shape_length - length;
   }
```

Comment:
the `shape_length` needs only be computed only when `idx < 0`
so something like:
```
if (idx < 0){
//compute shape_length
lengths[idx] = shape_length - length; //if (shape_length > length) not required
}
```

Reply:
shape[split_dim] - length